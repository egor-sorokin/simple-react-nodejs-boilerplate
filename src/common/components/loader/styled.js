import styled from 'styled-components';


export const SpinnerContainer = styled.div`
    position: absolute;
    top: 50%;
    left: 50%;
    
    transform: translateY(-50%);
`;

export const CenteredSpinnerContainer = styled.div`
    height: 120px;
    width: 100%;
    
    position: relative;
`;
