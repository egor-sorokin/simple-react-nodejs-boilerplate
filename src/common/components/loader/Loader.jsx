import React from 'react';
import Spinner from '@atlaskit/spinner';

import { SpinnerContainer } from './styled';


export const CenteredSpinner = () => (
    <SpinnerContainer><Spinner size="large"/></SpinnerContainer>
);
