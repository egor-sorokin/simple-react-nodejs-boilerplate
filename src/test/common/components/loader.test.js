import { shallow, render, mount } from 'enzyme';
import * as React from 'react'
import { CenteredSpinner } from '../../../common/components/loader/Loader';


describe('<CenteredSpinner />', () => {
    let component;

    describe('base tests', () => {
        beforeEach(() => {
            component = shallow(<CenteredSpinner />);
        });


        it('renders without crashing', () => {
            expect(component).toBeDefined();
        });


        it('matches snapshot', () => {
            expect(component).toMatchSnapshot();
        });


        it('renders 1 component', () => {
            expect(component.children()).toHaveLength(1);
        });
    });
});
