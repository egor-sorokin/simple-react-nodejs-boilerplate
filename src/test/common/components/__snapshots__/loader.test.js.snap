// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenteredSpinner /> base tests matches snapshot 1`] = `
<styled__SpinnerContainer>
  <Spinner
    delay={100}
    invertColor={false}
    isCompleting={false}
    onComplete={[Function]}
    size="large"
  />
</styled__SpinnerContainer>
`;
